packages/adapter-drizzle/src/lib/mysql.ts

```diff
@@ -2,263 +2,282 @@ import { and, eq } from "drizzle-orm"
+    providerAccountId: varchar("providerAccountId", { length: 255 }).notNull(),
+    refresh_token: varchar("refresh_token", { length: 255 }),
+    access_token: varchar("access_token", { length: 255 }),
+    expires_at: int("expires_at"),
+    token_type: varchar("token_type", { length: 255 }),
+    scope: varchar("scope", { length: 255 }),
+    id_token: varchar("id_token", { length: 2048 }),
+    session_state: varchar("session_state", { length: 255 }),
+  },
+  (account) => ({
+    compositePk: primaryKey({
+      columns: [account.provider, account.providerAccountId],
+    }),
+    userIdIdx: index("Account_userId_index").on(account.userId),
+  })
+)
+
+export const mysqlSessionsTable = mysqlTable(
+  "session" as string,
+  {
+    id: varchar("id", { length: 255 })
+      .primaryKey()
+      .$defaultFn(() => randomUUID()),
+    sessionToken: varchar("sessionToken", { length: 255 }).notNull().unique(),
     userId: varchar("userId", { length: 255 })
       .notNull()
-      .references(() => users.id, { onDelete: "cascade" }),
+      .references(() => mysqlUsersTable.id, { onDelete: "cascade" }),
     expires: timestamp("expires", { mode: "date" }).notNull(),
+  },
+  (session) => ({
+    userIdIdx: index("Session_userId_index").on(session.userId),
   })
+)
 
-  const verificationTokens = mySqlTable(
-    "verificationToken",
-    {
-      identifier: varchar("identifier", { length: 255 }).notNull(),
-      token: varchar("token", { length: 255 }).notNull(),
-      expires: timestamp("expires", { mode: "date" }).notNull(),
-    },
-    (vt) => ({
-      compoundKey: primaryKey(vt.identifier, vt.token),
-    })
-  )
-
-  return { users, accounts, sessions, verificationTokens }
-}
-
-export type DefaultSchema = ReturnType<typeof createTables>
+export const mysqlVerificationTokensTable = mysqlTable(
+  "verificationToken" as string,
+  {
+    identifier: varchar("identifier", { length: 255 }).notNull(),
+    token: varchar("token", { length: 255 }).notNull().unique(),
+    expires: timestamp("expires", { mode: "date" }).notNull(),
+  },
+  (vt) => ({
+    compositePk: primaryKey({ columns: [vt.identifier, vt.token] }),
+  })
+)
 
-export function mySqlDrizzleAdapter(
-  client: InstanceType<typeof MySqlDatabase>,
-  tableFn = defaultMySqlTableFn
+export function MySqlDrizzleAdapter(
+  client: MySqlDatabase<QueryResultHKT, PreparedQueryHKTBase, any>,
+  schema: DefaultMySqlSchema = {
+    usersTable: mysqlUsersTable,
+    accountsTable: mysqlAccountsTable,
+    sessionsTable: mysqlSessionsTable,
+    verificationTokensTable: mysqlVerificationTokensTable,
+  }
 ): Adapter {
-  const { users, accounts, sessions, verificationTokens } =
-    createTables(tableFn)
+  const { usersTable, accountsTable, sessionsTable, verificationTokensTable } =
+    schema
 
   return {
-    async createUser(data) {
-      const id = crypto.randomUUID()
+    async createUser(data: Omit<AdapterUser, "id">) {
+      const id = randomUUID()
```

Comment:
could we change this so we can leverage drizzle's `.defaultFn()` here and let users have control of the id generation?

Reply:
you have for the sessions table so don't see why users table should be any different

Reply:
ref https://github.com/nextauthjs/next-auth/pull/10479/files#r1573284149

Reply:
Hi, @juliusmarminge. As MySQL doesn't have `returning` clause, generated `id` was used to find the row after insertion. But you are right, we can find inserted row by `email` for user and by `sessionToken` for session and remove redundant `id` generation in the code.

Reply:
@realmikesolo will you remove these redundant ID's in your follow up PR as well?

Reply:
yep @ndom91